src/arrow_parquet/parquet_writer.rs

```diff
@@ -104,10 +104,12 @@ impl ParquetWriterContext {
 
         RecordBatch::try_new(schema, attribute_arrays).expect("Expected record batch")
     }
+}
 
-    pub(crate) fn close(self) {
+impl Drop for ParquetWriterContext {
+    fn drop(&mut self) {
         PG_BACKEND_TOKIO_RUNTIME
-            .block_on(self.parquet_writer.close())
+            .block_on(self.parquet_writer.finish())
```

Comment:
was waiting for that one. non-consuming version of `close` came for `parquet_writer` by arrow. That way we rely on drop.